R/data.table.R

```diff
@@ -467,7 +467,7 @@ replace_dot_alias = function(e) {
       # If using secondary key of x, f__ will refer to xo
       if (is.na(which)) {
         w = if (notjoin) f__!=0L else is.na(f__)
-        return( if (length(xo)) fsort(xo[w], internal=TRUE) else which(w) )
+        return( if (length(xo) && notjoin) fsort(xo[w], internal=TRUE) else which(w) )
```

Comment:
What's the `!length(xo)` case? When does that happen

Reply:
don't ask me, I haven't really understood that piece, I put browser and explore there to find something meaningful, I haven't, but I spotted `which(w)` was good answer already

Reply:
hmm but `xo` is something that I see a lot within `[.data.table`, just forgetting offhand what it means. one of the outputs of `forder`?

Reply:
Oh, this I know. if you do `x[xo]` you get sorted x.

Reply:
but when would `xo` be empty? just when `i` is empty?

Reply:
probably when `x` is already sorted

Reply:
OK, I think the fix basically makes sense then? also not 100% confident but it doesn't look like it would break anything